### DISCUSSION TITLE
Update docker image eclipse temurin

### DISCUSSION THREAD

**Discussion by 4soft-liebig** [renovatebot/renovate#16903] [Request Help] [answered]
Hi,
we are using a self-hosted instance of renovate bot to update a dockerfile that rests on eclipse temurin.
```FROM eclipse-temurin:17.0.3_7-jre-focal```
The `-focal` suffix is required, as this specific version has the OS/Arch support we need. The latest version is `17.0.4_8-jre-focal`; see https://hub.docker.com/_/eclipse-temurin/?tab=tags&page=1&name=17.0.4.

According to the logs the latest version is not considered:

```
       "config": {
         "dockerfile": [
           {
             "packageFile": "Dockerfile",
             "deps": [
               {
                 "depName": "eclipse-temurin",
                 "currentValue": "17.0.3_7-jre-focal",
                 "replaceString": "eclipse-temurin:17.0.3_7-jre-focal",
                 "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "depType": "final",
                 "depIndex": 0,
                 "updates": [],
                 "warnings": [],
                 "versioning": "docker",
                 "currentVersion": "17.0.3_7",
                 "fixedVersion": "17.0.3_7-jre-focal"
               }
             ]
           }
         ],
```

We are running renovate/renovate:32.137.0 on our self-hosted gitlab instance.

What can we do?
jl

**Comment by viceice** [comment, marked as answer]
use custom regex versioning[^1]

[^1]: https://docs.renovatebot.com/modules/versioning/#regular-expression-versioning

**Reply by 4soft-liebig** [reply]
Thanks for the fast response. The following package-rule using regular-expression-versioning works as expected:

```
    {
      "matchPackageNames": ["eclipse-temurin"],
      "versioning": "regex:^(?<major>\\d+)(\\.(?<minor>\\d+))?(\\.(?<patch>\\d+))?(_(?<build>\\d+))?(-(?<compatibility>.*))?$"
    }
```
